src/xor_space/prefix.rs

```diff
@@ -120,6 +120,15 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
         }
     }
 
+    /// Compares the prefixes using breadth-first order. That is, shorter prefixes are ordered
+    /// before longer. This is in contrast with the default `Ord` impl of `Prefix` uses depth-first
+    /// order.
+    pub fn cmp_breadth_first(&self, other: &Self) -> Ordering {
```

Comment:
just curious, why this different ordering is needed?

Reply:
This is because the function `create_connected_nodes_until_split` takes as the input a list of prefix lengths. This is then turned into actual prefixes which we want the network to eventually consist of and that is turned into a sequence of prefixes to split in order to get there. And it's this final sequence that needs to be ordered using this breadth-first order. I was considering changing the `Ord` impl instead, but eventually decided against it as I wasn't sure it would not break something.